src/main/java/io/leonis/subra/game/data/PlayerMeasurements.java

```diff
@@ -0,0 +1,47 @@
+package io.leonis.subra.game.data;
+
+import io.leonis.subra.game.data.Player.PlayerIdentity;
+import java.util.Map;
+import lombok.Value;
+
+/**
+ * The Class PlayerMeasurements.
+ *
+ * This class represents a collection of measurements for a {@link PlayerIdentity specific robot}.
+ *
+ * @author Rimon Oz
+ * @author Jeroen de Jong
+ */
+public interface PlayerMeasurements {
+
+  /**
+   * @return The identity of the player.
+   */
+  PlayerIdentity getPlayerIdentity();
+
+  /**
+   * @return Measurements identified by a unique string.
+   */
+  Map<String, Double> getMeasurements();
+
+  /**
+   * The Interface Supplier.
+   *
+   * Supplies a {@link PlayerMeasurements}.
+   *
+   * @author Jeroen de Jong
+   */
+  interface Supplier {
+
+    /**
+     * @return The measurements of a specific player.
+     */
+    PlayerMeasurements getPlayerMeasurements();
+  }
+
+  @Value
+  class State implements PlayerMeasurements {
```

Comment:
shouldn't this be public, and static too?

Reply:
As defined in [§9.1.1](https://docs.oracle.com/javase/specs/jls/se7/html/jls-9.html#jls-9.3) of the language spec, every field declaration in [...] an interface is implicitly public, static and final.